components/style/properties/longhand/box.mako.rs

```diff
@@ -1144,39 +1145,453 @@ ${helpers.predefined_type("scroll-snap-coordinate",
+                        try!(ay.to_css(f));
+                    f.write_str(")")
+                }
+                SpecifiedSkew::SkewX(ref ax) => {
+                    try!(f.write_str("skewX("));
+                        try!(ax.to_css(f));
+                    f.write_str(")")
+                }
+                SpecifiedSkew::SkewY(ref ay) => {
+                    try!(f.write_str("skewY("));
+                        try!(ay.to_css(f));
+                    f.write_str(")")
+                }
+            }
+        }
+    }
+
+    impl ToCss for SpecifiedSkew {
+        fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+            write!(dest, "{}", self)
+        }
+    }
+
+    #[derive(Clone, Debug, PartialEq)]
+    #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+    pub enum SpecifiedTranslate {
+        Translate(specified::LengthOrPercentage,
+                  specified::LengthOrPercentage),
+        TranslateX(specified::LengthOrPercentage),
+        TranslateY(specified::LengthOrPercentage),
+        TranslateZ(specified::Length),
+        Translate3D(specified::LengthOrPercentage,
+                    specified::LengthOrPercentage,
+                    specified::Length),
+    }
+
+    impl fmt::Display for SpecifiedTranslate {
+        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+            match *self {
+                SpecifiedTranslate::Translate(ref tx, ref ty) => {
+                    try!(f.write_str("translate("));
+                        try!(tx.to_css(f));
+                        try!(f.write_str(", "));
+                        try!(ty.to_css(f));
+                    f.write_str(")")
+                }
+                SpecifiedTranslate::TranslateX(ref tx) => {
+                    try!(f.write_str("translateX("));
+                        try!(tx.to_css(f));
+                    f.write_str(")")
+                }
+                SpecifiedTranslate::TranslateY(ref ty) => {
+                    try!(f.write_str("translateY("));
+                        try!(ty.to_css(f));
+                    f.write_str(")")
+                }
+                SpecifiedTranslate::TranslateZ(ref tz) => {
+                    try!(f.write_str("translateZ("));
+                        try!(tz.to_css(f));
+                    f.write_str(")")
+                }
+                SpecifiedTranslate::Translate3D(ref tx, ref ty, ref tz) => {
+                    try!(f.write_str("translate3d("));
+                        try!(tx.to_css(f));
+                        try!(f.write_str(", "));
+                        try!(ty.to_css(f));
+                        try!(f.write_str(", "));
+                        try!(tz.to_css(f));
+                    f.write_str(")")
+                }
+            }
+        }
+    }
+
+    impl ToCss for SpecifiedTranslate {
+        fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+            write!(dest, "{}", self)
```

Comment:
Yeah, in general I think we don't use `fmt::Display`. Just make them implement `ToCss` should be enough.